src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -76,7 +78,12 @@ class KNN80DocValuesConsumer extends DocValuesConsumer implements Closeable {
     public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
         delegatee.addBinaryField(field, valuesProducer);
         if (isKNNBinaryFieldRequired(field)) {
-            addKNNBinaryField(field, valuesProducer);
+            StopWatch stopWatch = new StopWatch();
```

Comment:
any reason for using stopwatch and not using the System.currentTimeInMillis/System.currentNano functions?

Reply:
System.currentTimeInMillis() doesn't always return the correct time.  Both StopWatch and System.nanoTime() work for calculating elapsed time, and in fact StopWatch uses System.nanoTime.  They're basically the same, but StopWatch has a few extra methods.